static/app/views/traces/fieldRenderers.spec.tsx

```diff
@@ -0,0 +1,315 @@
+    act(() => ProjectsStore.loadInitialData(projects));
+
+    MockApiClient.addMockResponse({
+      url: `/organizations/${organization.slug}/projects/`,
+      body: projects,
+    });
+  });
+
+  describe('SpanDescriptionRenderer', function () {
+    it('renders op then description', function () {
+      const span = makeSpan(projects[0]);
+
+      render(<SpanDescriptionRenderer span={span} />);
+
+      const description = screen.getByTestId('span-description');
+      expect(description).toBeInTheDocument();
+      expect(description).toHaveTextContent('op\u2014descriptioninternal_error');
+    });
+
+    it.each(['unknown', 'foobar'])(
+      'does not render span status %s',
+      function (spanStatus) {
+        const span = makeSpan(projects[0], {'span.status': spanStatus});
+
+        render(<SpanDescriptionRenderer span={span} />);
+
+        const description = screen.getByTestId('span-description');
+        expect(description).toBeInTheDocument();
+        expect(description).toHaveTextContent('op\u2014description');
+      }
+    );
+
+    it.each([
+      'ok',
+      'cancelled',
+      'invalid_argument',
+      'deadline_exceeded',
+      'not_found',
+      'already_exists',
+      'permission_denied',
+      'resource_exhausted',
+      'failed_precondition',
+      'aborted',
+      'out_of_range',
+      'unimplemented',
+      'internal_error',
+      'unavailable',
+      'data_loss',
+      'unauthenticated',
+    ])('renders span status %s', function (spanStatus) {
+      const span = makeSpan(projects[0], {'span.status': spanStatus});
+
+      render(<SpanDescriptionRenderer span={span} />);
+
+      const description = screen.getByTestId('span-description');
+      expect(description).toBeInTheDocument();
+      expect(description).toHaveTextContent(`op\u2014description${spanStatus}`);
+    });
+  });
+
+  describe('ProjectsRenderer', function () {
+    it('renders one project', function () {
+      render(<ProjectsRenderer projectSlugs={[projects[0].slug]} />, context);
+      expect(screen.getAllByRole('img')).toHaveLength(1);
+    });
+
+    it('renders two projects', function () {
+      render(
+        <ProjectsRenderer projectSlugs={[projects[0].slug, projects[1].slug]} />,
+        context
+      );
+      expect(screen.getAllByRole('img')).toHaveLength(2);
+    });
+
+    it('renders three projects', function () {
+      render(<ProjectsRenderer projectSlugs={projects.map(p => p.slug)} />, context);
+      expect(screen.getAllByRole('img')).toHaveLength(1);
+      const collapsed = screen.getByTestId('collapsed-projects-badge');
+      expect(collapsed).toBeInTheDocument();
+      expect(collapsed).toHaveTextContent('+2');
+    });
+  });
+
+  describe('ProjectRenderer', function () {
+    it('renders project badge without name', function () {
+      render(<ProjectRenderer projectSlug={projects[0].slug} hideName />);
+      expect(screen.getByRole('img')).toBeInTheDocument();
+    });
```

Comment:
You're not actually asserting the lack of the name here. Generally `not` is a bad idea, you can decorate with a test-id if you really want to test this empty state. If not I'd just delete this test.